tests/unit/components/preprint-status-banner-test.js

```diff
@@ -0,0 +1,203 @@
+import Ember from 'ember';
+import { moduleForComponent } from 'ember-qunit';
+import test from 'ember-sinon-qunit/test-support/test';
+import tHelper from 'ember-i18n/helper';
+
+
+moduleForComponent('preprint-status-banner', 'Unit | Component | preprint status banner', {
+    // Specify the other units that are required for this test
+    // needs: ['component:foo', 'helper:bar'],
+    unit: true,
+    needs: [
+        'model:action',
+        'model:node',
+        'model:user',
+        'model:preprint',
+        'model:preprint-provider',
+        'model:institution',
+        'model:comment',
+        'model:contributor',
+        'model:citation',
+        'model:file-provider',
+        'model:registration',
+        'model:draft-registration',
+        'model:log',
+        'model:taxonomy',
+        'model:license',
+        'model:wiki',
+        'service:i18n',
+        'service:theme',
+        'service:session',
+        'service:head-tags',
+    ],
+});
+
+
+test('getClassName computed property', function(assert) {
+    this.inject.service('store');
+    const component = this.subject();
+
+    Ember.run(() => {
+        const node = this.store.createRecord('node', {
+            title: 'test title',
+            description: 'test description',
+        });
+
+        const provider = this.store.createRecord('preprint-provider', {
+            reviewsWorkflow: 'pre-moderation',
+        });
+
+        const submission = this.store.createRecord('preprint', { node, provider });
+        component.setProperties({ submission });
+        component.set('reviewsWorkflow', 'pre-moderation');
+
+        component.set('submission.reviewsState', 'pending');
+        assert.strictEqual(component.get('getClassName'), 'preprint-status-pending-pre');
+
+        component.set('submission.reviewsState', 'accepted');
+        assert.strictEqual(component.get('getClassName'), 'preprint-status-accepted');
+    });
+});
+
+// Stub i18n service
+const i18nStub = Ember.Service.extend({
```

Comment:
Are translations unavailable during tests? Why do you need to stub `i18n`?

Reply:
Since this is a unit test for the component and we have extended i18n, I think it's actually a good idea to stub i18n here so that these tests don't rely on our extension of i18n working correctly. If this were an integration test, I would agree that it should just use our extended i18n.